dkulib/parallelizer/parallelizer.py

```diff
@@ -0,0 +1,298 @@
+            containing the `function` responses and errors. Default is "output".
+            This should be overriden by the developer: if the function to apply calls an API for text translation,
+            a good output_column_prefix would be "api_translation".
+        verbose: If True, log raw details on any error encountered along with the error message and error type.
+            Else (default) log only the error message and the error type.
+            We recommend trying without verbose first. Usually, the error message is enough to diagnose the issue.
+    """
+    # Default number of worker threads to use in parallel - may be tuned by the end user
+    DEFAULT_PARALLEL_WORKERS = 4
+    # By default, we assume the function to apply is row-by-row - should be overriden in the batch case
+    DEFAULT_BATCH_SUPPORT = False
+    # Default number of rows in one batch - may be tuned by the end user
+    DEFAULT_BATCH_SIZE = 10
+    # Default response parsing function for batch_size=1 - Simply assigns the response to the response column
+    DEFAULT_RESPONSE_PARSER = _parse_batch_response_default
+    # Default prefix to add to output columns - should be overriden for personalized output
+    DEFAULT_OUTPUT_COLUMN_PREFIX = "output"
+    # Default dictionary of output column names (key) and their descriptions (value)
+    OUTPUT_COLUMN_NAME_DESCRIPTIONS = OrderedDict(
+        [
+            ("response", "Raw response in JSON format"),
+            ("error_message", "Error message"),
+            ("error_type", "Error type or code"),
+            ("error_raw", "Raw error"),
+        ]
+    )
+    # By default, set verbose to False assuming error message and type are enough information in the logs
+    DEFAULT_VERBOSE = False
+
+    def __init__(
+        self,
+        function: Callable[[Union[Dict, List[Dict]]], Union[Dict, List[Dict]]],
+        error_handling: ErrorHandling = ErrorHandling.LOG,
+        exceptions_to_catch: Tuple[Exception] = (),
+        parallel_workers: int = DEFAULT_PARALLEL_WORKERS,
+        batch_support: bool = DEFAULT_BATCH_SUPPORT,
+        batch_size: int = DEFAULT_BATCH_SIZE,
+        batch_response_parser: Callable[[List[Dict], Any, NamedTuple], List[Dict]] = DEFAULT_RESPONSE_PARSER,
+        output_column_prefix: AnyStr = DEFAULT_OUTPUT_COLUMN_PREFIX,
+        verbose: bool = DEFAULT_VERBOSE,
+    ):
+        self.function = function
+        self.error_handling = error_handling
+        self.exceptions_to_catch = exceptions_to_catch
+        if error_handling == ErrorHandling.LOG and not exceptions_to_catch:
+            raise ValueError(
+                "Please set at least one exception in exceptions_to_catch")
+        self.parallel_workers = parallel_workers
+        self.batch_support = batch_support
+        if not batch_support:
+            # Overwrite necessary args for row-by-row iteration
+            batch_size = 1
+            batch_response_parser = _parse_batch_response_default
```

Comment:
You are setting the default for `batch_response_parser` a second time in case of batch support. If i understand well it will only have an impact in the unlikely case of someone puting `batch_support=False, batch_response_parser=somefunc`

Reply:
This will be redundant if we solve the point above

Reply:
@arnaudde that is correct. Should we remove this safeguard?

Reply:
I think we can remove it yes

Reply:
Do you want to remove the safeguard for batch_response_parser and/or the safeguard for batch_size?
I understand the rationale for removing it for batch_response_parser. 
Isn't it a bit weird to allow users to set batch_support=False and batch_size=10?

Reply:
I would keep the safeguard for `batch_size` and remove the one for `batch_response_parser`

Reply:
done in 7a1d214